src/arr/base/contracts.arr

```diff
@@ -33,13 +40,17 @@ data FieldFailure:
   | missing-field(loc, field) with:
     render-reason(self, loc, from-fail-arg):
       [ED.error:
-        [ED.para: ED.text("Missing field"), ED.code(ED.text(self.field)),
-          ED.text("is required at"), draw-and-highlight(self.loc)]]
+        [ED.para: ED.text("Missing field "), ED.code(ED.text(self.field)),
```

Comment:
I was hoping to spare you that level of nitpicky fixes :)  If you're willing to take that on, ok...  So in your branch, what does the `ErrorDisplay` now mean?  Looks like you have `paragraph`s, `v-sequence`s and `h-sequence`s: what do they correspond to?  I'm all for improving the `ErrorDisplay` data type; it was a first-draft implementation on my part, and I'm not wedded to its current design (or, by the same reasoning, to `ValueSkeleton`).

I actually wasn't looking for an explicit `ED.spaced` constructor, but was looking for a default where weneverwould wind up with missing spaces unless we opted in to that.  IMHO having extra spaces in English text is less annoying than having too few...